### DISCUSSION TITLE
Restrict ID/Relay Server to admins or disable

### DISCUSSION THREAD

**Discussion by dinger1986** [rustdesk/rustdesk#721] [Q&A] [answered]
Is it possible to hide ID/Relay server or only allow admins to edit it? Same with passwords and resetting the ID?

**Comment by dinger1986** [comment]
I think this should be a feature, I guess it's been changed to a discussion for others to give their views on it?

**Comment by rustdesk** [comment]
Yes

**Comment by rustdesk** [comment]
I do have been thinking about this. Because AnyDesk does do this way, but I thought it is not convenent.

**Reply by dinger1986** [reply]
I don't think it should be convenient, in a corporate environment it shouldn't be anyway and for home users they would never know any difference. Would help with security. 

Especially if the password and key in the toml file are salted in some way, then no one could view them without prior information.

**Reply by rustdesk** [reply]
So I need to collect more ideas for different scenarios or use cases.

**Comment by rustdesk** [comment]
and improve password generation? e.g. two passwords? the random password (updated periodically) on main windows, and another password can only be set and access by admin. Will check out AnyDesk and TV.

**Reply by dinger1986** [reply]
yeah sounds good to me, think teamviewer is random all the time but you can set your own password as an admin

**Reply by rustdesk** [reply]
> think teamviewer is random all the time

Do you know how often is it updated and when?

**Reply by Cedric-Demarais** [reply]
Hi,
About Teamviewer, you have 3 ways to access a remote computer without user interraction:
- With random password, wich you can set in preference to: never change, change after every connexion or disabled. You can also generate a new one by pushing a button like in RustDesk.
- With one or multiples personnal password, e.g. one for the admin, one for the user, another for another IT company, etc.
- With Teamviewer account, you can activate easy access for ONE TV account after validation processus.

You can have all 3 enabled at the same time.

Without going that far, I think the best should be:
- random password management: never change/change after every connexion/disabled
- Multiple (at least 2) personals passwords wich no one can read or easily decode (even in toml files) for security reason.

and be able to have both set at the time so generating a new random password doesn't affect the personnal(s) password(s).

Minimal could be a config parameter to set in RustDesk.toml to hide the password field in the GUI. And personnal password store encrypted.

Actualy, the main problem is that if i set a personal password in RustDesk, the user can see and change it, then i loose unconditional access.
Clear password or weak encryption used in config files is also an issue and force to have different personnal password for every remote PC.

Thanks for your great work, i think i'll leave TV as soon as i have a secure solution about this.

**Reply by rustdesk** [reply]
Thanks

**Reply by Cedric-Demarais** [reply]
By the way, a config parameter in RustDesk.toml to allow to hide the server key in the about windows and store it encryoted in config files would be nice because anyone can read it and can use my server without my consent (I already added -k _ to the command line to block unencrypted connections)
Also i think i'll not be able to detect them as i am using dockerized RustDesk and i don't know where to find log files of connections if any ?

Thanks :)

**Reply by rustdesk** [reply]
Windows TPM etc to save password more securely.

**Reply by 21pages** [reply]
|                                          | teamviewer                               | anydesk                                  | todesk                                   |
| ---------------------------------------- | ---------------------------------------- | ---------------------------------------- | ---------------------------------------- |
| Default                                  | Random password, Update at startup       | No password, confirm manually            | Temporary password, update manually      |
| Type of temporary password               | Random password                          | QR code                                  | Random/custom temporary password         |
| Temporary password updating              | After Each Session:<br/>1. Keep current<br/>2. Generate new<br/>3. Deactivate<br/>4. Show Confirmation to choose 1,2,3 above | via QR code                              | 1. Everyday<br/>2. Per session<br/>3. Random manually  <br/>4. Specific manually |
| Permanent(personal/preset/secure) password | Personal password                        | Preset password                          | Secure password                          |
| No password required                     | same account                             | not support                              | not support                              |
| UAC for password setting                 | No                                       | Yes                                      | No                                       |
| Password saving                          | 1. The controlled end can choose whether the control end can save the connection password temporarily, which means saving only at this startup. | 1. The controlled end can choose whether the control end can save the connection password<br/>2. The controlled end can delete the password saved by all control ends<br/>3. The control end can choose whether to save the connection password | The control end can choose whether to save the connection password |
| Passwords with different permission profiles | No                                       | There are three popular types of Permission Profiles:<br/>**Default**, **Screen Sharing**, **Full Access**,and **Unattended Access** for version compatible.<br/>Each has default permissions and can be customized | No                                       |
| Disable temporary password               | 1. Set disabled<br/>2. Optional disable aftere session<br/>3. Optional one-time password | Disabled by default                      | Set Disabled                             |
| Disable safe password                    | 1. Never set<br/>2. Clear strong password | Disabled by default                      | 1. Disabled by default<br/>2. Set Disabled |
| Other                                    | 1. Optional **one-time** password: disable random password access after session<br/>2. Keep the temporary password when the network is temporarily disconnected<br/>3. Fingerprint | 1. Optional **two-factor** authentication: preset password && QR code | 1. Optional **any** authentication: secure password \|\| temporary password |

**Reply by 21pages** [reply]
Password should be crypted:

security password
peer password
peer rdp password #891 
socks5 password
os password
address book password

**Reply by simeononsecurity** [reply]
The password complexity is too low by default. Roughly 6 characters and all lowercase.
I think there should be a change there. If we want to keep a short password minimum, then at least include upper, special and numbers.
Even if we block a system after multiple failed attempts, with only six characters it's mathematically possible to guess the connection password and still possible to brute force.

Options for disabling the password, not removing it, to allow for only admin set password would be nice as well.

**Comment by rustdesk** [comment]
![image](https://user-images.githubusercontent.com/71636191/176585934-6906b886-72fd-404a-a373-8d81ab989cd5.png)

Seems by default TeamViewer does not require administrative perm to modify global config options.

**Reply by simeononsecurity** [reply]
That may be true, doesn't mean that is smart however.
Keep in mind the windows default is to have all users on a system to be administrators as well..
This renders that setting useless unless it is in an enterprise.
In an enterprise setting, users don't have admin permissions and with a client like this allowing remote access from anywhere we really shouldn't be allowing end users to be able to see or the connection info.
This is a remote administration tool. It's not for gaming or for productivity. There are better options for that already. Unprivileged users shouldn't even know the software is installed let alone configure it.

**Comment by rustdesk** [comment]
@21pages let us discuss this (not password part, just  about who can modify settings) and  #1092 togther tomorrow.

**Reply by 21pages** [reply]
Teamviewer use UAC and global password to protect it's options, but it is not enabled by default. AnyDesk use UAC to protect some settings, including security, connection, recording, wake-on-lan. Todesk has no protection. 
The default account of windows has administrator privileges, so no password is required when UAC. Global password may be more effective on windows.

**Comment by paspo** [comment]
What about no password?
We already have asymmetric keys for connections, we can find a way to use (different!) keys to identify admins. That way we need only the public keys in client settings. And a way to safely store the private key for each machine.

**Reply by simeononsecurity** [reply]
Keys are great and all but it increases configuration complexity. Not all configurations warrant that level of security honestly. If rust desk were to use unsigned key pairs for authentication, it would be much smarter to just have that be the backend and have some sort of account to pair them to. Otherwise the appropriate enterprise alternative is active directory based logins or with full PKI and signed asymmetrical key pairs.

**Reply by paspo** [reply]
> 



> Otherwise the appropriate enterprise alternative is active directory based logins or with full PKI and signed asymmetrical key pairs.

I thought of that, but seemed overkill...

for now.

In the perfect (business w/ active directory) world we have rustdesk deployed via active directory and managed via its own admx template, but this will somewhat require active directory integration.

**Reply by rustdesk** [reply]
I talked to a company today who requires AD integration.

**Reply by AlexZigante** [reply]
For universities like where i work we would also appreciate/almost require AD integration. But as stated before locking down some key settings also for nonadmin users.

**Comment by rustdesk** [comment]
Correction: We have never defined RustDesk as a remote administration tool. We need to comprimise among different scenarios. Or we may publish different version for different scenario.

**Reply by simeononsecurity** [reply]
Your competition all call themselves remote administration tools. You call yourself an alternative to your competitors. 
Not to mention that all your users are already using it for remote administration of systems.
It's already typecast, regardless of what you call it. A shiny turd is still a turd. A Lambo is still fast even if you drive it slow. 
A DVD is still a disk even if you use it as a coaster.

**Reply by simeononsecurity** [reply]
It's best to adapt to your primary use case. Or the use case that can bring the most potential enterprise users and support. Rn, IMPO is an open source remote administration tool.

Capabilities for gaming or general productivity distract from that and there are better options that you can t compete with for those.

Other than ssh, rdp, and vnc, which don't really accomplish the same thing at the same level and the closed source options like TV and anydesk you're the only option.

To grab market share, you need active directory integration, fully self hosted support, FIPS 140-2 compliance, and a general security minded focus.

**Reply by rustdesk** [reply]
Good point.

**Comment by mrwrighty** [comment]
Unattended access needs to have a hidden password as in Anydesk. We install Anydesk on Client PC's (They know its there) it saves time reaching out for support as we use the same password for our clients. When support is requested we just need the ID unless we have connected previously. New users requiring support download Anydesk and run it on the fly. I then ask to install it and I set an unattended password which I do not want them to see.

**Reply by lukasrajzl** [reply]
In my company we need attended acces, some of our users work with sensible data and our it-security dont allow any tools that have unattended acces. All remote app must have config for attended acces. Only user can allow remote login.

**Reply by mrwrighty** [reply]
Thats fine don't set the unattended access and the user only runs RustDesk when they require support and they give you the ID and new access password when IT need access.

**Reply by lukasrajzl** [reply]
How do you not set unattended acces?  Password wont change automaticaly - so if I connect to client one time and save his password and second time I have unattended access. This is what our IT-security dont allow. I need that password is automaticaly change after end of session or user must always click on accept connection.

**Comment by hboetes** [comment]
Perhaps a set of admin configured options would be nice.

The settings below I'd like to set for all my users, so they don't have to configure them, and can't change them:
```
rendezvous_server = 'meet.example.com:21116'
nat_type = 2
serial = 0

[options]
enable-audio = 'N'
approve-mode = 'click'
custom-rendezvous-server = 'meet.example.com'
key = 'foobar='
```

- enable-audio: This can make the connection quite a bit slower, and I do not want to hear the music my colleagues listen to.
- approve-mode: Remote access without the VPN, seems like a bad idea.
- rendezvous_server: Not something my colleagues should need to configure.
- key: same.

**Reply by dinger1986** [reply]
Can you not just roll the toml file out to all users?

**Reply by hboetes** [reply]
That doesn't prevent them from overwriting the defaults, after which they can set a password and use it for remote access. And there are a lot of users and only one admin.

**Comment by rustdesk** [comment, marked as answer]
Done in nightly build.

**Reply by rustdesk** [reply]
for installed only

**Reply by dinger1986** [reply]
amazing!!

**Comment by studio37hu** [comment]
@rustdesk There are quite a lot of small IT companies (like ours) that supports their customers using such programs like RustDesk – we are not allowed in these cases to restrict (local PC) users on the remote clients. Therefore we really need the option in RustDesk: "Lock settings with password." In this case (local PC admin) users are allowed to receive connections, but can not view, copy or modify ID/Relay server or Key. This function is essential to finally be able to setup our own server, since we do not want to risks that some users start using our servers without our knowledge for any purpose.

**Reply by vlad-sinyakov** [reply]
I agree with you, such functionality would be very useful

**Reply by studio37hu** [reply]
> I agree with you, such functionality would be very useful

In fact, since version 1.3.0, we can lock important settings (security and network) with a PIN code, which is perfectly enough to prevent the end user from peeking. :) See Settings - Security - Unlock with PIN (the last option).

**Reply by vlad-sinyakov** [reply]
Wow, didn't see this option in the menu. Thanks.
